### PR TITLE
doc: matter: Fixes for the Matter documentation

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
- Update of release table
- Removed BUTTON 4 description from Lock sample
- Added missing Matter Settings shell commands section

TODO:

- [x] Update Hardware requirements
- [x] Check links
- [x] Check samples' README
- [x] Check rendered new pages.

### CODE REVIEW

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst [hunk 1]
```diff
@@ -64,6 +64,28 @@ This option lets you use the Matter shell commands with :ref:`matter_samples`.
...
 
+.. _ug_matter_configuring_settings_shell:
+
+Matter Settings shell commands
+------------------------------
+
+You can enable the Matter Settings shell commands to monitor the current usage of the Zephyr Settings NVS.
+These commands are useful for verifying that the ``settings`` partition has the proper size and meets the application requirements.
+
+To enable Matter Settings shell module set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL` Kconfig option to ``y``.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:75]:
```suggestion
To enable the Matter Settings shell module set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL` Kconfig option to ``y``.
```

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst [hunk 2]
```diff
@@ -64,6 +64,28 @@ This option lets you use the Matter shell commands with :ref:`matter_samples`.
...
+
+You can use the following shell commands:
+
+* ``matter_settings peak`` - Read the maximum settings usage peak.
+* ``matter_settings reset`` - Reset the peak value.
+* ``matter_settings get_size <name>`` - Get size of the specific entry.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:81]:
```suggestion
* ``matter_settings get_size <name>`` - Get the size of the specific entry.
```

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst [hunk 3]
```diff
@@ -64,6 +64,28 @@ This option lets you use the Matter shell commands with :ref:`matter_samples`.
...
+* ``matter_settings current`` - Get size of the current settings usage.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:82]:
```suggestion
* ``matter_settings current`` - Get the size of the current settings usage.
```

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst [hunk 4]
```diff
@@ -64,6 +64,28 @@ This option lets you use the Matter shell commands with :ref:`matter_samples`.
...
+* ``matter_settings free`` - Get size of the current free settings space.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:83]:
```suggestion
* ``matter_settings free`` - Get the size of the current free settings space.
```

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst [hunk 5]
```diff
@@ -64,6 +64,28 @@ This option lets you use the Matter shell commands with :ref:`matter_samples`.
...
+
+.. note::
+
+  Matter Settings shell module is available only for the NVS Zephyr Settings backend.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:87]:
```suggestion
  The Matter Settings shell module is available only for the NVS Zephyr Settings backend.
```

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 6]
```diff
@@ -57,107 +57,95 @@ Values are provided in kilobytes (KB).
       The following table lists memory requirements for samples running on the :ref:`nRF52840 DK <programming_board_names>` (:ref:`nrf52840dk_nrf52840 <zephyr:nrf52840dk_nrf52840>`).
 
       +------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
-      | Sample                                                           |   MCUboot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
+      | Sample                                                           |   MCUBoot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:60]:
```suggestion
      | Sample                                                           |   MCUboot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
```

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 7]
```diff
@@ -57,107 +57,95 @@ Values are provided in kilobytes (KB).
...
+      | :ref:`Window Covering <matter_window_covering_sample>` (Release) |            28 |               630 |              4 |         32 |         694 |                             161 |
       +------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
 
    .. tab:: nRF5340 DK
 
       The following table lists memory requirements for samples running on the :ref:`nRF5340 DK <programming_board_names>` (:ref:`nrf5340dk_nrf5340_cpuapp <zephyr:nrf5340dk_nrf5340>`).
 
       +------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
-      | Sample                                                           |   MCUboot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
+      | Sample                                                           |   MCUBoot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:88]:
```suggestion
      | Sample                                                           |   MCUboot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
```

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 8]
```diff
@@ -57,107 +57,95 @@ Values are provided in kilobytes (KB).
...
+      | :ref:`Window Covering <matter_window_covering_sample>` (Release) |            32 |               555 |              4 |         32 |         623 |                             165 |
       +------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
 
    .. tab:: Thingy:53
 
       The following table lists memory requirements for samples running on the :ref:`Thingy:53 <programming_board_names>` (:ref:`thingy53_nrf5340 <zephyr:thingy53_nrf5340>`).
 
       +---------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
-      | Sample                                                        |   MCUboot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
+      | Sample                                                        |   MCUBoot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:116]:
```suggestion
      | Sample                                                        |   MCUboot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
```

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 9]
```diff
@@ -57,107 +57,95 @@ Values are provided in kilobytes (KB).
...
+      | :ref:`Weather Station <matter_weather_station_app>` (Release) |            64 |               574 |             16 |         48 |         702 |                             188 |
       +---------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
 
    .. tab:: nRF7002 DK
 
       The following table lists memory requirements for samples running on the :ref:`nRF7002 DK <programming_board_names>` (:ref:`nrf7002dk_nrf5340_cpuapp <nrf7002dk_nrf5340>`).
 
       +------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
-      | Sample                                                     |   MCUboot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
+      | Sample                                                     |   MCUBoot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:128]:
```suggestion
      | Sample                                                     |   MCUboot ROM |   Application ROM |   Factory data |   Settings |   Total ROM |   Total RAM (incl. static HEAP) |
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 10]
```diff
@@ -679,19 +680,17 @@ Matter samples
 
   * Fixed an issue that prevented nRF Toolbox for iOS in version 5.0.9 from controlling the sample using :ref:`nus_service_readme`.
   * Changed the design of the :ref:`matter_lock_sample_wifi_thread_switching` feature so that support for both Matter over Thread and Matter over Wi-Fi is included in a single firmware image.
+  * Added support for Wi-Fi firmware upgrade on external memory, only for the combination of the nRF5340 DK with the nRF7002 EK.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:683]:
```suggestion
  * Added support for Wi-Fi firmware upgrade on external memory.
    This is only available for the nRF5340 DK with the nRF7002 EK.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst (hunk 8 above)
Comment (ArekBalysNordic) [reply doc/nrf/protocols/matter/getting_started/hw_requirements.rst:116]: Hmm, So I will update the script for generating the data also :)

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst (hunk 9 above)
Comment (ArekBalysNordic) [reply doc/nrf/protocols/matter/getting_started/hw_requirements.rst:128]: The tables will be updated later once again. I need to re-generate data.

File: doc/nrf/protocols/matter/index.rst [hunk 11]
```diff
@@ -20,9 +20,9 @@ The |NCS| allows you to develop applications with different versions of Matter,
 +--------------------------+-----------------------------------------------------+------------------------+
 | nRF Connect SDK version  | Matter specification version                        | `Matter SDK version`_  |
 +==========================+=====================================================+========================+
-| v2.5.99 (latest)         | :ref:`1.2.0 <ug_matter_overview_dev_model_support>` | 1.2.0.1                |
+| |release|                | :ref:`1.2.0 <ug_matter_overview_dev_model_support>` | 1.2.0.1                |
 +--------------------------+-----------------------------------------------------+------------------------+
-| |release|                | :ref:`1.1.0 <ug_matter_overview_dev_model_support>` | 1.1.0.1                |
+| v2.5.2                   | :ref:`1.1.0 <ug_matter_overview_dev_model_support>` | 1.1.0.1                |
```
Comment (ArekBalysNordic) [inline doc/nrf/protocols/matter/index.rst:25]: @wiba-nordic Is it appropriate regarding the NCS 2.6.0 release?

Comment (wiba-nordic) [reply doc/nrf/protocols/matter/index.rst:25]: Yes, I neede to think about this for a bit myself. After the release, we'll re-add an entry for latest.

Comment (ArekBalysNordic) [reply doc/nrf/protocols/matter/index.rst:25]: I mean, this should replace itself with 2.6.0 automatically after the release tag, am I right?

Comment (wiba-nordic) [reply doc/nrf/protocols/matter/index.rst:25]: Correct.

File: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst [hunk 12]
```diff
@@ -64,6 +64,28 @@ This option lets you use the Matter shell commands with :ref:`matter_samples`.
...
 
+.. _ug_matter_configuring_settings_shell:
+
+Matter Settings shell commands
+------------------------------
+
+You can enable the Matter Settings shell commands to monitor the current usage of the Zephyr Settings NVS.
+These commands are useful for verifying that the ``settings`` partition has the proper size and meets the application requirements.
+
+To enable the Matter Settings shell module set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL` Kconfig option to ``y``.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst:75]:
```suggestion
To enable the Matter Settings shell module, set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL` Kconfig option to ``y``.
```

**Review by LuDuda — COMMENTED** [review]
This is awesome! Some nits on memory maps.

**Review by doublemis1 — APPROVED** [review]
LGTM!

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 13]
```diff
@@ -123,45 +131,51 @@ Values are provided in kilobytes (KB).
...
+      | :ref:`Thermostat <matter_thermostat_sample>` (Debug)              |            48 |               886 |              4 |         32 |         970 |                             266 |
+      +-------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
+      | :ref:`Thermostat <matter_thermostat_sample>` (Release)            |            48 |               786 |              4 |         32 |         870 |                             258 |
+      +-------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
 
 ..
 
+.. note::
+  The results for ``Link-Time Optimization`` based on building the Thread library from sources.
+  If you use the binary Thread library version the ROM usage may be higher.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:177]:
```suggestion
  The results for ``Link-Time Optimization`` are based on building the Thread library from sources.
  If you use the binary Thread library version, the ROM usage may be higher.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst [hunk 14]
```diff
@@ -56,65 +56,73 @@ Values are provided in kilobytes (KB).
...
+      +-------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
+      | :ref:`Template <matter_template_sample>` (Debug)                  |            32 |               626 |              4 |         32 |         694 |                             171 |
+      +-------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
+      | :ref:`Template <matter_template_sample>` (Release)                |            32 |               530 |              4 |         32 |         598 |                             164 |
+      +-------------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+---------------------------------+
+      | :ref:`Template <matter_template_sample>` (Link-Time Optimization)*|            32 |               486 |              4 |         32 |         554 |                             244 |
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:116]:
```suggestion
      | :ref:`Template <matter_template_sample>` (Link-Time Optimization)*|            32 |               486 |              4 |         32 |         554 |                             244 |
```
The asterisk does not work in the table, and is breaking the doc build.

**Review by wiba-nordic — APPROVED** [review]
Everything looks to be in order.

**Review by LuDuda — APPROVED** [review]

**Pull request merged**